src/reducers/appReducer.js

```diff
@@ -2,14 +2,28 @@ import {
   ADD_APP_ERROR,
   REMOVE_APP_ERROR,
   SET_CURRENT_EDITOR,
+  TOGGLE_WELCOME,
+  CHANGE_WELCOME_SEARCH_VALUE,
+  CHANGE_WELCOME_SEARCH_RESULTS,
+  TOGGLE_WELCOME_SEARCH_LOADING,
 } from '../actions/constants';
 
+
+
 const initialState = {
   error: null,
   id: null,
   editor: 'contract',
+  welcome: {
+    toggle: true,
+    searchValue: '',
+    results: [],
+    isLoading: false,
+  }
```

Comment:
Is there a reason you are using redux for the welcome page state instead of the local state of the component? I'm not sure this needs to be in the redux store.

Reply:
i think i am using redux just for the sake of being consistent with the code base, having looked through the code i thought that you mainly use redux for storing states, that's why i did that too, but if you prefer in this case to store it locally, I can change that

Reply:
For this component, I'd say local state makes the most sense and would be a little less complex. If you could change it that would be great, thanks!